cooked-validators/src/Cooked/Attack/DupToken.hs

```diff
@@ -1,59 +1,51 @@
+{-# LANGUAGE FlexibleContexts #-}
+
 module Cooked.Attack.DupToken where
 
+import Control.Monad
+import Cooked.Attack.AddConstraints
 import Cooked.Attack.Common
 import Cooked.MockChain.Wallet
-import Cooked.Tx.Constraints
 import Cooked.Tx.Constraints.Optics
-import Data.Maybe
 import qualified Ledger as L
 import Optics.Core
-import qualified PlutusTx.Numeric as Pl
 
--- | A token duplication attack which modifies every 'Mints'-constraint of a
--- 'TxSkel' that satisfies some conditions. This adds a 'DupTokenLbl' to the
--- labels of the transaction using 'addLabel'.
+-- | A token duplication attack increases values in 'Mints'-constraints of a
+-- 'TxSkel' according to some conditions, and pays the extra minted value to a
+-- given recipient wallet. This adds a 'DupTokenLbl' to the labels of the
+-- transaction using 'addLabel'. Returns the 'Value' by which the minted value
+-- was increased.
 dupTokenAttack ::
   -- | A function describing how the amount of minted tokens of an asset class
-  -- should be changed by the attack. The given function @f@ should probably satisfy
-  -- > f ac i == Just j -> i < j
-  -- for all @ac@ and @i@, i.e. it should increase in the minted amount.
-  -- If it does *not* increase the minted amount, or if @f ac i == Nothing@, the
-  -- minted amount will be left unchanged.
-  (L.AssetClass -> Integer -> Maybe Integer) ->
+  -- should be changed. The given function @f@ should satisfy @f ac i > i@ for
+  -- all @ac@ and @i@, i.e. it should increase the minted amount. If it does
+  -- *not* increase the minted amount, the amount will be left unchanged.
+  (L.AssetClass -> Integer -> Integer) ->
```

Comment:
Maybe it is worth explaining that the `AssetClass` and the `Integer` parameter in this function correspond to the `AssetClass` and the amount of a token in a Mints constraint.